mitodl/mit-open
# Clean up resource descriptions

### What are the relevant tickets?

Closes https://github.com/mitodl/hq/issues/4355

### Description (What does it do?)
Strips away any HTML tags in the `description` or `full_description` fields that don't match a list of allowed tags.


### How can this be tested?
- Run `./manage.py backpopulate_prolearn_data`
- Go to http://localhost:8063/search?offered_by=mitpe&q=%22Leadership+and+Innovation%22, click on the 1 result.
- The drawer description should look like this:

<img width="590" alt="Screenshot 2024-05-28 at 3 43 47 PM" src="https://github.com/mitodl/mit-open/assets/187676/0ea9adef-5d55-4c8b-9748-9084165de46e">


and definitely not like this:


<img width="590" alt="Screenshot 2024-05-28 at 3 43 00 PM" src="https://github.com/mitodl/mit-open/assets/187676/2125e80c-8bcf-4c5b-a771-c4132136967f">


### Additional Context
I decided not to include links as allowed HTML tags @Ferdi, let me know if you feel otherwise.

`full_description` is only present (and different from `description`) for edx courses, so I got rid of the code that assigns it the same value as `description` for some of the other sources.
